### ISSUE TITLE
batch dimension in eval.py

### ISSUE THREAD

**Issue by dmitrysarov** [milesial/Pytorch-UNet#41]
https://github.com/milesial/Pytorch-UNet/blob/0f45521480d2de8a93407ce43a2a813eefd5d1f6/eval.py#L22
Is it right:
unsqueeze(0) for true_mask emulate batch dimension?
but then taken only net(img)**[0]** for mask_pred
Why?
For my task i have 2 channels in mask, and removing  **[0]** did a job.
Just want to understand.

**Comment by milesial** [comment]
The following `dice_coeff(mask_pred, true_mask)` call should take two batched elements, so I shouldn't have taken `net(img)[0]`. It makes sense that it works by removing the `[0]`.

I don't know how it was working for me, it seems the code is wrong, but it worked with one channel in the mask (not 100% sure about that).

**Comment by dmitrysarov** [comment]
my only assumption that it could be because of flat mask , when mask shape is (h,w), not (c=1,h,w)

**Comment by milesial** [comment]
Yeah the dice loss function was designed withthe behaviour of n_classes=1 in mind, so it might be different when n_classes > 1.

**Comment by microshiftd** [comment]
I have the same problem, when n_classes = 2, the dice loss can not converge
